network-api/networkapi/mozfest/templates/partials/mixed_content_card.html

```diff
@@ -0,0 +1,40 @@
+{% load wagtailcore_tags wagtailimages_tags %}
+
+{% image item.image fill-720x405 as img %}
+<div class="tw-m-0 tw-p-0">
+    <div class="tw-flex tw-flex-col tw-gap-6 tw-w-full">
+        {% if item.link_page or item.link_url %}
+            <a href="{% if item.link_page %}{% pageurl item.link_page %}{% elif item.link_url %}{{ item.link_url }}{% endif %}" aria-label="{{ item.title }}">
+        {% endif %}
+            <img width="720" height="405" src="{{ img.url }}" alt="{{ item.alt_text }}"/>
+        {% if item.link_page or item.link_url %}
+            </a>
+        {% endif %}
+
+        {% if item.highlighted_metadata or item.metadata %}
+            <div class="tw-flex tw-flex-wrap">
+                {% if item.highlighted_metadata %}
+                    <span class="tw-h6-heading tw-text-blue-80 tw-mr-4 tw-py-2 tw-mb-0">{{ item.highlighted_metadata }}</span>
+                {% endif %}
+
+                {% if item.metadata %}
+                    <span class="tw-h6-heading tw-text-gray-40 tw-py-2 tw-mb-0">{{ item.metadata }}</span>
+                {% endif %}
+            </div>
+        {% endif %}
+
+        {% if item.link_page or item.link_url %}
```

Comment:
Yea, that is true. 

I think this would be valuable still, since there already is (and has been for a while) template and block overload in the project. The proliferation of templates that are basically the same but used to handle different data source make this project particularly hard to work on. 

I think it would be great if there is any chance we can get this in with a BE dev. 

@chris-lawton Can you add some more description what exactly was not working for you with the existing card template so that we can address that with the BE work?

Reply:
[generic_card.html](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/generic_card.html) expects `{% pageurl page %}` so that needs modifying to be not required (as mixed content cards can be without a link) and accept either an internal or external url for both instances where it's used. We also need to render out a different size image and add a line clamp to the body text.

Reply:
Turns out the refactor is actually not that big if you keep the page as a possible parameter: https://github.com/MozillaFoundation/foundation.mozilla.org/pull/11601